tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Golden testing in Haskell with Hspec-Golden
+published: 2016-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golden-1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: A golden test is an IO action that writes its result to a file. To pass the test, this output file should be identical to the corresponding golden file, which contains the correct result for the test.
+---
+
+Golden tests are like unit tests, with the difference that the expected output is stored in a separate file. 
+
+Golden tests can also be known as:
+* Golden master testing
+* Approval testing
+* Snapshot testing
+* Expect testing
+
+There are many forms of getting the same result with different approaches, as the [Jest](https://jestjs.io/docs/en/snapshot-testing.html) snapshot testing,
+comparing two screenshots as [Wraith](https://github.com/BBC-News/wraith) or with the [Approval testing method](https://dzone.com/articles/approval-tests-alternative).
+
+##The Main idea
+A golden test is just an **IO action** that writes its result to a file.
+To pass the test, the output **file** should be identical to the corresponding golden file,
+that contains the correct result for the given test.
```

Comment:
@MedAdrian Probably, we're being redundant with this section. Aren't we covering this in the introduction?

Reply:
I've removed this section.